server/index.js

```diff
@@ -32,22 +32,43 @@ app.use('/api/users', userRoutes)
 app.use('/api/posts', postRoutes)
 app.use('/api/upload', uploadRoute)
 
-mongoose.connect(process.env.MONGO_URI, { useNewUrlParser: true, useUnifiedTopology: true }, () => {
-  console.log("Connected to database")
+const dbConnect = async () => new Promise((resolve) => {
+  mongoose.connect(process.env.MONGO_URI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    serverSelectionTimeoutMS: 10000,
+  }, (err) => {
+    if (err) {
+      throw err
+    }
```

Comment:
Got a bit confused with why things weren't working until I realised MongoDB wasn't connecting. This should make those errors a little clearer.